core/src/test/java/apoc/refactor/GraphRefactoringTest.java

```diff
@@ -994,5 +998,148 @@ public void testRefactorCategorizeNoDups() {
         assertEquals(8, relsCount);
         db.executeTransactionally("DROP CONSTRAINT ON (n:`" + label + "`) ASSERT n.`" + targetKey + "` IS UNIQUE");
     }
+
+    @Test
+    public void testMergeNodeShouldNotCreateSelfRelationshipsInPreExistingSelfRel() {
+        db.executeTransactionally("CREATE (a:TestNode {a:'a'})-[:TEST_REL]->(b:TestNode {a:'b'})-[:TEST_REL]->(c:TestNode {a:'c'})\n" +
+                "WITH a, c CREATE (a)-[:TEST_REL]->(a) WITH c CREATE (c)-[:TEST_REL]->(c);");
+        testCall(db, "MATCH (n:TestNode) WITH collect(n) as nodes CALL apoc.refactor.mergeNodes(nodes, {mergeRels: true, selfRel: false}) yield node return node",
```

Comment:
There should probably be a distinction between:

1. existing self-rels
2. creating new self-rels when you merge

The default current named `selfRel` should probably be renamed to `produceSelfRel` or something similar and account only for the self-rels described in point 2.
Self-rels in point 1 should by default be preserved. Maybe another config flag like `preserveExistingSelfRels` could be added (with default value `true`)?

Reply:
@fbiville Good idea, I considered only the new `selfRel`,
leaving unchanged the previous behavior, that is to cancel the existing relationships. 
Now I changed `selfRel` to `produceSelfRel` and added the second one property.